nilearn/input_data/nifti_spheres_masker.py

```diff
@@ -138,6 +200,65 @@ def __call__(self, imgs):
         return signals, None
 
 
+def _spheres_inversion(seeds_, radius, mask_img, allow_overlap):
+    '''Utility function to create a maps Niimg, containing a map with a sphere
+    of the given radius for each seed_ location.
+
+    Parameters
+    ----------
+    seeds_: List of triplets of coordinates in native space
+        Seed definitions. List of coordinates of the seeds in the same space
+        as target_affine (either of mask or if mask is None MNI space).
+    radius: float
+        Indicates, in millimeters, the radius for the sphere around the seed.
+    mask_img: Niimg-like object
+        NiftiSpheresMasker mask_img, is necessary as reference to back-project 
+        spheres to brain-space.
+    allow_overlap: boolean
+        If False, an error is raised if the maps overlaps (ie at least two
+        maps have a non-zero value for the same voxel).
+
+    Returns
+    -------
+    spheres_map: Niimg-like object
+        A map of each of the sphere's locations. Scaled at places of overlap.
+    '''
+    # Mask is necessary for affine and shape
+    if mask_img is not None:
+        mask = check_niimg_3d(mask_img)
+    else:
+        raise ValueError('Please provide mask_img at initialization to'
+                            ' provide a reference for the inverse_transform')
+
+    flat_mask, _ = masking._load_mask_img(mask_img)
+    flat_mask = flat_mask.reshape(-1) 
+
+    n_seeds = len(seeds_)
+
+    spheres = np.zeros((np.prod(mask.shape), n_seeds))
```

Comment:
we should use a sparse matrix instead. but do we really need to store all at once, since we are iterating over seeds and treating all separately?

Reply:
I will give it a try - as I said I am not experienced with sparse matrices.